From 6142350d7c0562b0525462abdd41309ae0faa2eb Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 6 Aug 2021 19:49:37 +0200
Subject: [PATCH] module/system/base/target.scm: support riscv32

Fix the following build failure on riscv32:

system/base/target.scm:132:16: In procedure triplet-pointer-size:
unknown CPU word size "riscv32"

Fixes:
 - http://autobuild.buildroot.org/results/6705630c1484239ec8b73d57ebc2e2570fbfc8f8

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 module/system/base/target.scm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/module/system/base/target.scm b/module/system/base/target.scm
index 2088cd866..dba46664d 100644
--- a/module/system/base/target.scm
+++ b/module/system/base/target.scm
@@ -116,6 +116,7 @@
 
           ((string-match "^x86_64-.*-gnux32" triplet) 4)  ; x32
 
+          ((string-match "32$" cpu) 4)
           ((string-match "64$" cpu) 8)
           ((string-match "64_?[lbe][lbe]$" cpu) 8)
           ((member cpu '("sparc" "powerpc" "mips" "mipsel" "nios2" "m68k" "sh3" "sh4")) 4)
-- 
2.30.2

